pkg/treeoperator/controller/virtualcluster_join_controller.go

```diff
@@ -1,16 +1,452 @@
+	}
+
+	deleteRequest := types.NamespacedName{
+		Namespace: request.Namespace,
+		Name:      clustertreeDeploy.Name,
+	}
+	err = c.Get(ctx, deleteRequest, clustertreeDeploy)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("get clustertree deployment error, deployment deleted failed: %v", err)
+		}
+		klog.Infof("clustertree deployment doesn't exist")
+	} else {
+		err := c.Delete(ctx, clustertreeDeploy)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("delete kosmos-clustertree Deployment error: %v", err)
+		}
+	}
+
+	klog.Info("Deployment has been deleted. ")
+
+	klog.Info("Start deleting kosmos-clustertree secret")
+	clustertreeSecret, err := kosmosctl.GenerateSecret(treemanifest.ClusterTreeClusterManagerSecret, treemanifest.SecretReplace{
+		Namespace: request.Namespace,
+		Cert:      cert.GetCrtEncode(),
+		Key:       cert.GetKeyEncode(),
+		Name:      request.Name,
+	})
+	if err != nil {
+		return err
+	}
+	deleteRequest.Name = clustertreeSecret.Name
+	err = c.Get(ctx, deleteRequest, clustertreeSecret)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("get clustertree secret error, secret deleted failed: %v", err)
+		}
+		klog.Infof("clustertree secret doesn't exist")
+	} else {
+		err := c.Delete(ctx, clustertreeSecret)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("delete kosmos-clustertree Deployment error: %v", err)
+		}
+	}
+	klog.Info("Deployment has been deleted. ")
+
+	clusterName := fmt.Sprintf("virtualcluster-%s-%s", request.Namespace, request.Name)
+	klog.Infof("Attempting to delete cluster %s...", clusterName)
+
+	kubeconfigStream, err := base64.StdEncoding.DecodeString(vc.Spec.Kubeconfig)
+	if err != nil {
+		return fmt.Errorf("decode target kubernetes kubeconfig %s err: %v", vc.Spec.Kubeconfig, err)
+	}
+	err, kosmosClient, _, _ := c.InitTargetKubeclinet(kubeconfigStream)
+	if err != nil {
+		return fmt.Errorf("create kubernetes client failed: %v", err)
+	}
+
+	old, err := kosmosClient.KosmosV1alpha1().Clusters().Get(context.TODO(),
+		clusterName, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("get cluster %s failed when we try to del: %v", clusterName, err)
+		}
+	} else {
+		err = c.RemoveClusterFinalizer(old, kosmosClient)
+		if err != nil {
+			return fmt.Errorf("removefinalizer %s failed: %v", clusterName, err)
+		}
+		err = kosmosClient.KosmosV1alpha1().Clusters().Delete(context.TODO(), clusterName, metav1.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("delete cluster %s failed: %v", clusterName, err)
+		}
+	}
+	klog.Infof("Cluster %s has been deleted.", clusterName)
+	if err := c.RemoveFinalizer(ctx, vc); err != nil {
+		return fmt.Errorf("remove finalizer error: %v", err)
+	}
+	return nil
+}
+
+func (c *VirtualClusterJoinController) InitTargetKubeclinet(kubeconfigStream []byte) (error, versioned.Interface, kubernetes.Interface, extensionsclient.Interface) {
```

Comment:
There are misspellings of words ｜ InitTargetKubeClient

Reply:
fix it